packages/mdx/mdx-hast-to-jsx.js

```diff
@@ -29,8 +29,10 @@ function toJSX(node, parentNode = {}) {
       importNodes.map(childNode => toJSX(childNode, node)).join('\n') +
       '\n' +
       exportNodes.map(childNode => toJSX(childNode, node)).join('\n') +
+      (layout ? `function objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }` +
+      '\n' : '') +
       '\n' +
-      `export default ({components}) => <MDXTag name="wrapper" ${layout ? `Layout={${layout}}` : ''} components={components}>${jsxNodes
+      `export default (props) => <MDXTag name="wrapper" ${layout ? `Layout={${layout}} layoutProps={objectWithoutProperties(props, ['components'])}` : ''} components={props.components}>${jsxNodes
```

Comment:
Since the output JSX still requires babel parsing maybe it makes sense to use destructuring?

```js
export default ({ components, ...props }) =>
```

Reply:
Great idea! That way I won't be breaking anything.